Fingerprints Processing.py

```diff
@@ -0,0 +1,364 @@
+    # white background 
+    im[:] = 255 
+
+    for i in range(int(w/block_size)): 
+        for j in range(int(h/block_size)): 
+            angle = angles.item(j, i) 
+            if angle != 0: 
+                angle = -1/math.tan(math.radians(angle)) 
+                p1, p2 = get_line_ends(i, j, angle, block_size, 2) 
+                cv2.line(im, p1, p2, (0,0,255), 1) 
+    return im 
+ 
+def orientation(img, block_size, smooth=False): 
+    h, w = img.shape 
+
+    # make a reflect border frame to simplify kernel operation on borders 
+    borderedImg = cv2.copyMakeBorder(img, block_size,block_size,block_size,block_size, cv2.BORDER_DEFAULT) 
+
+    # apply a gradient in both axis 
+    sobelx = cv2.Sobel(borderedImg, cv2.CV_64F, 1, 0, ksize=3) 
+    sobely = cv2.Sobel(borderedImg, cv2.CV_64F, 0, 1, ksize=3) 
+    angles = np.zeros((int(h/block_size), int(w/block_size)), np.float32) 
+    for i in range(int(w/block_size)): 
+        for j in range(int(h/block_size)): 
+            nominator = 0. 
+            denominator = 0. 
+
+        # calculate the summation of nominator (2*Gx*Gy) 
+        # and denominator (Gx^2 - Gy^2), where Gx and Gy 
+        # are the gradient values in the position (j, i) 
+            for k in range(block_size): 
+                for l in range(block_size): 
+                    posX = block_size-1 + (i*block_size) + k 
+                    posY = block_size-1 + (j*block_size) + l 
+                    valX = sobelx.item(posY, posX) 
+                    valY = sobely.item(posY, posX) 
+
+                    nominator += f(valX, valY) 
+                    denominator += g(valX, valY) 
+
+            # if the strength (norm) of the vector  
+            # is not greater than a threshold 
+            if math.sqrt(nominator**2 + denominator**2) < 1000000: 
+                angle = 0. 
+            else: 
+                if denominator >= 0: 
+                    angle = cv2.fastAtan2(nominator, denominator) 
+                elif denominator < 0 and nominator >= 0: 
+                    angle = cv2.fastAtan2(nominator, denominator) + math.pi 
+                else: 
+                    angle = cv2.fastAtan2(nominator, denominator) - math.pi 
+                angle /= float(2) 
+    
+            angles.itemset((j, i), angle) 
+
+    if smooth: 
+        angles = cv2.GaussianBlur(angles, (3,3), 0, 0) 
+    return angles 
+ 
+def draw_orientation(h, w, angles, block_size): 
+    im = np.zeros((h, w), np.uint8) 
+    for i in range(int(w/block_size)): 
+        for j in range(int(h/block_size)):  
+            dangle = 2*angles.item(j, i) 
+            v = int(round(dangle * (255/float(360)))) 
+            for k in range(block_size): 
+                for l in range(block_size): 
+                    im.itemset((j*block_size+l,i*block_size+k), v) 
+    return im 
+
+    
+if __name__ == "__main__":
+    for i in sys.argv[1:3]:
+        input_folder = i
+        #normal
+        if i == sys.argv[1]:
```

Comment:
the command should be something like:
python processfp option param1 param2

and check the value of option, but the index of the argv for params should start from argv[3].

Reply:
make it possible to run one variation each time.